crates/pathfinder/src/state/sync.rs

```diff
@@ -330,24 +328,26 @@ async fn update_sync_status_latest(
     use crate::rpc::types::{BlockNumberOrTag, Tag};
     let poll_interval = head_poll_interval(chain);
 
+    let starting = NumberedBlock::from((starting_block_hash, starting_block_num));
+
     loop {
         match sequencer
             .block_by_number(BlockNumberOrTag::Tag(Tag::Latest))
             .await
         {
             Ok(block) => {
-                let latest_hash = block.block_hash.unwrap();
-                let latest_num = block.block_number.unwrap();
+                let latest = {
+                    let latest_hash = block.block_hash.unwrap();
+                    let latest_num = block.block_number.unwrap();
```

Comment:
I cannot see any regression test for the `latest_num` being updated. I know it seems a bit silly because that was an easy fix, but we still don't have an automated way of making sure it won't happen again in the future.

Generally I recall seeing at least another fix in the past to a regression that was not accompanied by a respective test. I think we should put more stress on tests when regressions pop up.